get_trace_data.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+import argparse
+import multiprocessing
+import requests
+import os
+import pickle
+import json
+from typing import Callable
+import mapillary
+
+OUTPUT_DIR = 'output'
+TEMP_DIR = 'tmp'
+SECTIONS_PICKLE_FILENAME = 'sections.pickle'
+MAX_FILES_IN_DIR = 500  # Maximum number of files we will put in one directory
+
+
+def initialize_dirs(bbox: str) -> tuple[str, str]:
+    """
+    Creates all dirs needed for run if they don't exist
+    :param bbox: bbox string from arg
+    :return: tuple of (output dir name, tmp dir name for any tmp pickle files)
+    """
+    output_dir = os.path.join(
+        os.getcwd(),
+        OUTPUT_DIR,
+        bbox
+    )
+    output_tmp_dir = os.path.join(
+        os.getcwd(),
+        OUTPUT_DIR,
+        bbox,
+        TEMP_DIR
+    )
+    # Make the output and tmp dirs if it does not exist yet
+    if not os.path.exists(output_tmp_dir):
+        os.makedirs(output_tmp_dir)  # Makes all dirs recursively, so we know output_dir will also now exist
+
+    return output_dir, output_tmp_dir
+
+
+def split_bbox(output_dir: str, bbox: str, to_bbox_str: Callable[[float, float, float, float], str],
+               section_size: float = 0.05) -> list[str]:
+    """
+    Takes the given bbox and splits it up into smaller sections, with the smaller bbox chunks having long/lat sizes =
+    section_size. Also writes the bbox sections to disk so we can pick up instructions from previous runs (may be
+    removed)
+    :param output_dir: output dir name
+    :param bbox: bbox string from arg
+    :param to_bbox_str: function that takes (min_long, min_lat, max_long, max_lat) bbox definition coordinates, and
+    returns a string that we will feed into the next function. Should be the same format as the API source expects
+    :param section_size: the smaller bbox sections will have max_long-min_long = max_lat-min_lat = section_size
+    :return: list of bbox section strings, whose format will be dictated by the to_bbox_str function
+    """
+    sections_filename = os.path.join(output_dir, SECTIONS_PICKLE_FILENAME)
+
+    try:
+        print('Reading bbox_sections from disk...')
+        bbox_sections = pickle.load(open(sections_filename, 'rb'))
+    except (OSError, IOError):
+        print('bbox_sections pickle not found. Creating and writing to disk...')
+        min_long, min_lat, max_long, max_lat = [float(s) for s in bbox.split(',')]
+
+        # Perform a check to see how many sections would be generated
+        num_files = ((max_long - min_long) // section_size + 1) * ((max_lat - min_lat) // section_size + 1)
+        if num_files > MAX_FILES_IN_DIR:
+            # TODO: Check len of bbox_sections, if over some size limit, we split things up
```

Comment:
sounds good

Reply:
Just a note here, I elected not to implement this part yet because it makes some of the filesystem layout and logic a bit messy. So right now it just prints out a warning if there's going to be a ton of files generated.